packages/grafana-runtime/src/services/pluginExtensions/extensions.test.ts

```diff
@@ -69,6 +69,89 @@ describe('getPluginExtensions', () => {
   });
 });
 
+describe('deepFreeze()', () => {
+  test('should not fail when called with primitive values', () => {
+    expect(deepFreeze(1)).toBe(1);
+    expect(deepFreeze('foo')).toBe('foo');
+    expect(deepFreeze(true)).toBe(true);
+    expect(deepFreeze(false)).toBe(false);
+    expect(deepFreeze(undefined)).toBe(undefined);
+    expect(deepFreeze(null)).toBe(null);
+  });
+
+  test('should freeze an object so it cannot be overriden', () => {
+    const obj = {
+      a: 1,
+      b: '2',
+      c: true,
+    };
+    const frozen = deepFreeze(obj);
+
+    expect(() => {
+      frozen = { a: 234 };
+    }).toThrow(TypeError);
```

Comment:
nit: Maybe use `Object.isFrozen()` for these assertions rather than checking if a type error occurs? I could be wrong but my thinking here is that checking for thrown type errors doesn't necessarily assert the `deepFreeze` function is doing what I think it is. 🤔

Reply:
I think you are right bud, I should probably check that as well, thanks 👍

(I would probably keep both checks, as _ideally_ that line should only throw a type error if the property cannot be changed due to being frozen, and it would be a bit harder to test if an object is deeply frozen like that - I guess at least.)

Reply:
Addressed in https://github.com/grafana/grafana/pull/64964/commits/3cf173c1f80f2be871588bb97f02e4dd49dfea09.